### ISSUE TITLE
[BUG] NLTK's PunktTokenizer fails to initialize due to UnicodeDecodeError [Windows specific]

### ISSUE THREAD

**Issue by soras** [nltk/nltk#3294]
Recently, PunktTokenizer's data loading has been changed to [load data from tab files](https://github.com/nltk/nltk/pull/3283); however, the change introduces `UnicodeDecodeError` on Windows platform, where the default (file) encoding is not 'utf-8'. 

Code example:
```python
>>> from nltk.tokenize.punkt import PunktTokenizer
>>> base_sentence_tokenizer = PunktTokenizer(lang='estonian')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Programmid\Miniconda3\envs\py311_devel\Lib\site-packages\nltk\tokenize\punkt.py", line 1744, in __init__
    self.load_lang(lang)
  File "C:\Programmid\Miniconda3\envs\py311_devel\Lib\site-packages\nltk\tokenize\punkt.py", line 1750, in load_lang
    self._params = load_punkt_params(lang_dir)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Programmid\Miniconda3\envs\py311_devel\Lib\site-packages\nltk\tokenize\punkt.py", line 1770, in load_punkt_params
    params.ortho_context = pdec.tab2intdict(f)
                           ^^^^^^^^^^^^^^^^^^^
  File "C:\Programmid\Miniconda3\envs\py311_devel\Lib\site-packages\nltk\tabdata.py", line 108, in tab2intdict
    return defaultdict(int, {a: int(b) for a, b in self.tab2tups(f)})
                                                   ^^^^^^^^^^^^^^^^
  File "C:\Programmid\Miniconda3\envs\py311_devel\Lib\site-packages\nltk\tabdata.py", line 50, in tab2tups
    return [self.tab2tup(rm_nl(x)) for x in f]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Programmid\Miniconda3\envs\py311_devel\Lib\site-packages\nltk\tabdata.py", line 50, in <listcomp>
    return [self.tab2tup(rm_nl(x)) for x in f]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Programmid\Miniconda3\envs\py311_devel\Lib\encodings\cp1257.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'charmap' codec can't decode byte 0x9a in position 423: character maps to <undefined>
```
Same error appears with `PunktTokenizer(lang='finnish')`. 

It seems that the problem can be fixed if parameter  `encoding='utf-8'` is passed to `open` here: 
https://github.com/nltk/nltk/tree/577d2f950a31f9f5bd5fbf7b5936b08318689983/nltk/tokenize/punkt.py#L1763-L1770
https://github.com/nltk/nltk/blob/develop/nltk/tokenize/punkt.py#L1763-L1770

**Comment by ekaf** [comment]
Thanks @soras, it is indeed necessary to pass _encoding='utf-8'_ on any system that doesn't have that as default. Could you please submit a PR with your suggested changes? This would ensure that that it can be reviewed quicker.

**Comment by ekaf** [comment]
In principle, we should expect this issue to be resolved, now that #3296 is merged.
